judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/action/relationGetRangeReferenceBehaviour.etl

```diff
@@ -144,4 +123,69 @@ rule CreateCardinalityForGetRangeRelationEntityTypeOutput
         t.lower = 0;
         t.upper = -1;
 }
+*/
+
+// rules for create get range input type
+
+@lazy
+@greedy
+rule CreateCardinalityForGetRangeRelationTransferTypeInputOwner
+    transform s: JSL!TransferRelationDeclaration
+    to t : JUDOPSM!Cardinality {
+        t.setId("(jsl/" + s.getId() + ")/CreateCardinalityForGetRangeRelationTransferTypeInputOwner");
+        t.lower = 0;
+        t.upper = 1;
+}
+
+@lazy
+@greedy
+rule CreateCardinalityForGetRangeRelationTransferTypeInputQueryCustomizer
+    transform s: JSL!TransferRelationDeclaration
+    to t : JUDOPSM!Cardinality {
+        t.setId("(jsl/" + s.getId() + ")/CreateCardinalityForGetRangeRelationTransferTypeInputQueryCustomizer");
+        t.lower = 0;
+        t.upper = 1;
+}
+
+@lazy
+@greedy
+rule CreateTransferObjectRelationForGetRangeRelationTransferTypeInputOwner
+    transform s: JSL!TransferRelationDeclaration
+    to t : JUDOPSM!TransferObjectRelation {
+        t.setId("(jsl/" + s.getId() + ")/GetRangeRelationTransferTypeInputOwner");
+        t.name = "owner";
+        t.target = s.eContainer.getTransferDeclarationEquivalent();
+        t.embedded = true;
+        t.cardinality = s.equivalent("CreateCardinalityForGetRangeRelationTransferTypeInputOwner");
+}
+
+@lazy
+@greedy
+rule CreateTransferObjectRelationForGetRangeRelationTransferTypeInputQueryCustomizer
+    transform s: JSL!TransferRelationDeclaration
+    to t : JUDOPSM!TransferObjectRelation {
+        t.setId("(jsl/" + s.getId() + ")/GetRangeRelationTransferTypeInputQueryCustomizer");
+        t.name = "queryCustomizer";
+        t.target = s.referenceType.equivalent("CreateQueryCustomizerType");
+        t.embedded = true;
+        t.cardinality = s.equivalent("CreateCardinalityForGetRangeRelationTransferTypeInputQueryCustomizer");
+}
 
+@lazy
+@greedy
+rule CreateGetRangeInputType
+    transform s: JSL!TransferRelationDeclaration
+    to t: JUDOPSM!UnmappedTransferObjectType {
+        t.setId("(jsl/" + s.getId() + ")/GetRangeInputType");
```

Comment:
name of the rule and the id postfix is different

Reply:
In my opinion, the ID will be long, but let it be

Reply:
[59ae12b](https://github.com/BlackBeltTechnology/judo-tatami-jsl/pull/314/commits/59ae12b4f85e82ed18322fc8fad4e90bb6bfd026)